fem/pbilinearform.cpp

```diff
@@ -246,6 +246,13 @@ void ParBilinearForm::Assemble(int skip_zeros)
    }
 }
 
+void ParBilinearForm::AssembleDiagonal(Vector &diag) const
+{
+   Vector local(pfes->GetVSize());
+   BilinearForm::AssembleDiagonal(local);
+   pfes->GetProlongationMatrix()->MultTranspose(local, diag);
```

Comment:
You should check the pointer returned by `pfes->GetProlongationMatrix()` -- it will be `NULL` when the prolongation is the identity, e.g. conforming mesh on one processor.

Reply:
This does not appear to actually be the case. `ParFiniteElementSpace::GetProlongationMatrix()` never returns `NULL`, and below in `ParBilinearForm::FormLinearSystem()` (which is what I modeled this method on) the value returned from `GetProlongationMatrix()` is used without any checking. I can certainly add checking but the current form seems consistent with the rest of the code.

Reply:
You can see the issue by running `./ex1p -pa -d cuda`.

Reply:
This is indeed an issue (with partial assembly the code that gets executed is `Operator::FormLinearSystem`)

Reply:
Fixed now, let me know what you think.